hr_timesheet_sheet/models/hr_timesheet_sheet.py

```diff
@@ -414,7 +416,7 @@ def action_timesheet_refuse(self):
     def button_add_line(self):
```

Comment:
This is called on a single timesheet only, this can be expressed as
```
@api.multi
def button_add_line(self):
    self.ensure_one()

    if self.state == 'draft':
       self.add_line()
```

w/o need to pass details

Reply:
This is called on a single timesheet only, this can be expressed as
```
@api.multi
def button_add_line(self):
    self.ensure_one()

    if self.state == 'draft':
       self.add_line()
```

w/o need to pass details

Reply:
I don't like to put "ensure ones" in "api multi" methods. Just my taste and preference.

Reply:
I got that, yet I don't see how exactly this method may be called for multiple objects?